java-security/src/main/java/com/sap/cloud/security/token/AbstractToken.java

```diff
@@ -162,7 +163,24 @@ public int hashCode() {
 
 	@Override
 	public String getZoneId() {
-		return getClaimAsString(TokenClaims.SAP_GLOBAL_ZONE_ID);
+		return getClaimAsString(SAP_GLOBAL_ZONE_ID);
+	}
+
+	@Override
+	public String getClientId() {
+		String clientId = getClaimAsString(AUTHORIZATION_PARTY);
+		if (clientId == null || clientId.trim().isEmpty()) {
+			Set<String> audiences = getAudiences();
+
+			if (audiences.size() == 1) {
+				return audiences.stream().findFirst().get();
+			} else {
+				LOGGER.error("Couldn't get client id. Invalid authorized party or audience claims.");
```

Comment:
check for cid (and add a comment that this is only required for support generated tokens in JUnit tests. Finally they should provide "azp" claim instead of "cid" claim) and then als log a WARNING